polygeist/test/polygeist-opt/sycl/cast.mlir

```diff
@@ -1,45 +1,45 @@
-// RUN: polygeist-opt --convert-polygeist-to-llvm --split-input-file %s | FileCheck %s
+// RUN: polygeist-opt --convert-polygeist-to-llvm='use-opaque-pointers=1' --split-input-file %s | FileCheck %s
```

Comment:
We need, for the time being, to have both the original test cases that use the default (non-opaque pointers) as well as the new one. The reason is that, until we switch, we want to ensure that code changes do not regress existing tests. 
WDYT ?

Reply:
Yes, good idea. I've added back the original tests without modifications for opaque pointers, with `-typed-pointer`-suffix to their name.